sharedbuffers/mapped_struct.py

```diff
@@ -1760,6 +1823,255 @@ def _c_search_hkey_ui32(hkey, pindex, stride0, length, hint):
                 return mid
         return lo
 
+    @cython.cfunc
+    @cython.locals(
+        hkey = cython.longlong,
+        mkey = cython.longlong, ikey = cython.int, imkey = cython.int,
+        lo = cython.size_t, hi = cython.size_t, length = cython.size_t,
+        mid = cython.size_t, mid2 = cython.size_t, stride0 = cython.size_t, hint = cython.size_t,
+        pindex = cython.p_char, skip = cython.size_t)
+    @cython.returns(cython.size_t)
+    def _c_search_hkey_i32(hkey, pindex, stride0, length, hint):
+        hi = length
+        lo = 0
+        ikey = hkey
```

Comment:
It's to avoid type casts within the loop, which would be hard on the compiler. Operations on matching types are easier to optimize (ie: they don't have to concern themselves with overflow for instance).

Other methods are uiXX (which use uikey, similar thing), and the float versions, which just use the correct type for hkey and don't need a typecasted local.